### PR TITLE
PR for Program and Entity classes 

### PR DESCRIPTION
**Pull request by pedraumks**
Program class:

Screen Shot 2019-10-05 at 8 52 28 PM

Steps to reproduce:
1) F5 to start debugger on the Program class
2) Add breakpoint at any line (line 25 in the screenshot)
3) Click launch program

Entity class:
Screen Shot 2019-10-05 at 9 00 46 PM

Steps to reproduce:
1) Add console.log statements for this.name, this.uniqueAttributes, and this.nonUniqueAttributes at the end of the parse function and uncomment the following three lines at the bottom of ENTITY.js
```
Tokenizer.makeTokenizer("entity.txt")
const x = new ENTITY();
x.parse();
```
2) Run `node src/ast/ENTITY.js`

### CODE REVIEW

File: src/ast/ENTITY.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            if (cur === "end") return;
+            // this means that non unique attributes are optional 
+            cur = this.tokenizer.getNext();
+            while (cur !== "end") {
+                this.nonUniqueAttributes.push (cur);
+                cur = this.tokenizer.getNext();
+            }
+            //console.log(this.uniqueAttributes);
+            //console.log(this.nonUniqueAttributes);
```
Comment (jasmingoh) [inline src/ast/ENTITY.js]: let's remove console log comments

File: src/ast/ENTITY.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        catch (err) {
+            throw new Error("Unable to build AST");
+        }
+    }
+
+    evaluate() {
+        //console.log(this.name);
+        //console.log(this.uniqueAttributes);
+        //console.log(this.nonUniqueAttributes);
```
Comment (jasmingoh) [inline src/ast/ENTITY.js]: here as well

**Comment by jasmingoh** [comment]
looks good! running as expected!

File: src/ast/PROGRAM.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        if (this.relationships.length != 0) {
+            for (let i = 0; i < this.relationships.length; i++) {
+                this.relationships[i].evaluate();
+            }
+        }
+    }
+}
+
+Tokenizer.makeTokenizer("sample.txt");
+const x = new PROGRAM();
```
Comment (theAriKh) [inline src/ast/PROGRAM.js]: This should be removed or commented out since they are for testing.

**Comment by ellenliu** [comment]
Can you not merge the files Relationship.js and Edge.js? i'm still making changes to those, dont want this version to be master

File: src/ast/ENTITY.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+const Node = require('../parser/Node')
+const Tokenizer = require('../parser/Tokenizer')
+
```
Comment (ellenliu) [inline src/ast/ENTITY.js]: missing `const ATTRIBUTE = require('./ATTRIBUTE')`

File: src/ast/ATTRIBUTE.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+const Node = require('../parser/Node')
+const Tokenizer = require('../parser/Tokenizer')
+
```
Comment (ellenliu) [inline src/ast/ATTRIBUTE.js]: missing `const EDGE = require('./EDGE')`

**Pull request merged**